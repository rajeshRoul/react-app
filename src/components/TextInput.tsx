import '../styles/TextInput.css';
import React from 'react';

type TextInputPropType = {
  type : string;
  placeholder : string;
  value : string;
  handleChange : any;
}

function TextInput(props : TextInputPropType) {
  const {
    type, placeholder, value, handleChange,
  } = props;
  return (
    <input className="TextInput" type={type} placeholder={placeholder} value={value} onChange={handleChange} />
  );
}

export default TextInput;
