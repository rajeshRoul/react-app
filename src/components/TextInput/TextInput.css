.TextInput {
    outline: none;
    font-size: 1.5rem;
    border: none;
    border-bottom: 2px solid black;
    margin-bottom: 30px;
    background-color: transparent;
}

.DarkTextInput{
    color: white
}

.DarkTextInput::placeholder{
    color: rgb(192, 192, 192);
}