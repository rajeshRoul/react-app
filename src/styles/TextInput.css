.TextInput {
    outline: none;
    font-size: 1.5rem;
    border: none;
    border-bottom: 2px solid black;
    margin-bottom: 30px;
}
