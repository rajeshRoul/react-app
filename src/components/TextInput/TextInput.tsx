import './TextInput.css'
import React, { ReactElement, useContext } from 'react'
import { ThemeContext } from '../../util/themeContext'

type TextInputPropType = {
    type: string
    placeholder: string
    value: string
    handleChange: any
}

function TextInput(props: TextInputPropType): ReactElement {
    const { type, placeholder, value, handleChange } = props
    const { theme } = useContext(ThemeContext)
    return (
        <input
            data-testid="textInput"
            className={`TextInput ${theme === 'Dark' && 'DarkTextInput'}`}
            type={type}
            placeholder={placeholder}
            value={value}
            onChange={handleChange}
        />
    )
}

export default TextInput
