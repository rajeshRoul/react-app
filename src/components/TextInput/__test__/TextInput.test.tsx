/* eslint-disable no-undef */
import React, { ReactElement } from 'react'
import { render, cleanup } from '@testing-library/react'
import TextInput from '../TextInput'
import '@testing-library/jest-dom/extend-expect'
import ThemeProvider, { ThemeContext } from '../../../util/themeContext'

const renderWithContext = (component: ReactElement): any => {
    return {
        ...render(
            <ThemeProvider value={ThemeContext}>{component}</ThemeProvider>
        ),
    }
}

afterEach(cleanup)

test('button click', () => {
    const { getByTestId } = renderWithContext(
        <TextInput
            type="text"
            placeholder="test placeholder"
            value="test"
            handleChange={() => {}}
        />
    )
    expect(getByTestId('textInput')).toHaveValue('test')
})

it('should take a snapshot', () => {
    const { getByTestId } = renderWithContext(
        <TextInput
            type="text"
            placeholder="test placeholder"
            value="test"
            handleChange={() => {}}
        />
    )
    expect(getByTestId('textInput')).toMatchSnapshot()
})
