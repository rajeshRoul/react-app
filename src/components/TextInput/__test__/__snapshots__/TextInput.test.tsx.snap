// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<input
  class="TextInput false"
  data-testid="textInput"
  placeholder="test placeholder"
  type="text"
  value="test"
/>
`;
